### PR TITLE
PRM sample

### PR DESCRIPTION
**Pull request by yinghany**
Add a new WDF driver sample to demonstrate the usage of the PRM direct-call interface.

### CODE REVIEW

**Review by JakobL-MSFT — CHANGES_REQUESTED** [review]
Sent email with details!

File: prm/samples/PrmFunc/prmfuncsample.vcxproj [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    <SampleGuid>{157FF815-CA85-49A2-B622-5D57EFAB0A13}</SampleGuid>
+    <WindowsTargetPlatformVersion>$(LatestTargetPlatformVersion)</WindowsTargetPlatformVersion>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup Label="Configuration" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <TargetVersion>Windows10</TargetVersion>
+    <UseDebugLibraries>False</UseDebugLibraries>
+    <DriverTargetPlatform>Desktop</DriverTargetPlatform>
+    <DriverType>KMDF</DriverType>
+    <PlatformToolset>WindowsKernelModeDriver11.0</PlatformToolset>
```
Comment (NeoAdonis) [inline prm/samples/PrmFunc/prmfuncsample.vcxproj:36]: This platform toolset does not exist.

File: prm/samples/PrmFunc/prmfuncsample.vcxproj [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    <DriverType>KMDF</DriverType>
+    <PlatformToolset>WindowsKernelModeDriver10.0</PlatformToolset>
+    <ConfigurationType>Driver</ConfigurationType>
+  </PropertyGroup>
+  <PropertyGroup Label="Configuration" Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
+    <TargetVersion>Windows10</TargetVersion>
+    <UseDebugLibraries>False</UseDebugLibraries>
+    <DriverTargetPlatform>Desktop</DriverTargetPlatform>
+    <DriverType>KMDF</DriverType>
+    <PlatformToolset>WindowsKernelModeDriver11.0</PlatformToolset>
```
Comment (NeoAdonis) [inline prm/samples/PrmFunc/prmfuncsample.vcxproj:52]: This platform toolset does not exist.

**Comment by NeoAdonis** [comment]
* Unless you expect to have more PRM samples, you should move the contents of prm/samples to simply prm to simplify folder structure.
* If possible, please add support for ARM64.

**Review by NeoAdonis — CHANGES_REQUESTED** [review]

**Review by NeoAdonis — APPROVED** [review]

**Comment by JakobL-MSFT** [comment]
@NeoAdonis : Sample still does not build!!  See build failure.

**Comment by yinghany** [comment]
> @NeoAdonis : Sample still does not build!! See build failure.

@JakobL-MSFT What is the WDK version that you are building with? This is a new sample reliant on Ge WDK headers (prminterface.h).

**Comment by JakobL-MSFT** [comment]
> @JakobL-MSFT What is the WDK version that you are building with? This is a new sample reliant on Ge WDK headers (prminterface.h).

Now I understand!!!  

First. You need to target our "develop" branch.  Not "main".

Second.  The samples in the develop branch needs to build with (1) "in market" EWDK and (2) also GE EWDK.  Exceptions can happen (as in your case) and we add them to .\exclusions.csv file in relevant branch.  The entry would look as follows:
---
prm\prmsample,*,,22621,Only NI: Not supported on NI.
---

(Please note the .\exclusion.csv file differs between main and develop branch.  You need to edit based on the develop branch version)

I can make the edit above (or you can add it - please keep the entries sorted)

Cheers.

Jakob

**Pull request closed without merging**